syntax/puppet.vim

```diff
@@ -1,167 +1,433 @@
-" puppet syntax file
-" Filename:     puppet.vim
-" Language:     puppet configuration file
-" Maintainer:   Luke Kanies <luke@madstop.com>
-" URL:
-" Last Change:
-" Version:
```

Comment:
On version and filename, fair enough. Last change is a convention for Vim files, even Vim's own files follow it.